### PR TITLE
RAB-1149: Enable authenticator manager

### PR DESCRIPTION
**Pull request by MadeWilson**
**Description (for Contributor and Core Developer)**

In this PR, we migrated the PIM security to use the new authenticator manager system (cf [Symfony blog post](https://symfony.com/blog/new-in-symfony-5-1-updated-security-system)).

So, we : 
- Upgraded akeneo/oauth-server-bundle to the new major which support authenticator manager (see https://github.com/akeneo/FOSOAuthServerBundle/pull/2)
- Updated `security.yml` to conforms to config changes and depreciations
- Updated our custom code to conform to the new API
- Fixed minor deprieciations (`UsernamePasswordToken` construction)

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: config/packages/security.yml [hunk 1]
```diff
@@ -17,20 +18,13 @@ security:
             pattern:                        ^/(_(profiler|wdt)|css|images|js)/
             security:                       false
 
-        install:
```
Comment (ghoussard) [inline config/packages/security.yml:20]: Unused

File: config/packages/security.yml [hunk 2]
```diff
@@ -77,12 +71,13 @@ security:
                 name:                       BAPRM
                 lifetime:                   1209600   # stay logged for two weeks
                 samesite:                   'lax'
-            anonymous:                      false
 
     access_control:
-        - { path: ^/admin/, role: ROLE_ADMIN }
```
Comment (ghoussard) [inline config/packages/security.yml:83]: Unused

**Review by StevenVAIDIE — APPROVED** [review]
There is a lot of remaining : 

```
new UsernamePasswordToken($user, null, 'main', $user->getRoles());
```

In code can you made another PR to fix it ?

**Review by ghoussard — APPROVED** [review]

**Comment by ghoussard** [comment]
> There is a lot of remaining :
> 
> ```
> new UsernamePasswordToken($user, null, 'main', $user->getRoles());
> ```
> 
> In code can you made another PR to fix it ?

I'll do it in another PR yes

**Pull request merged**